### PR TITLE
docs: mention about branch model (refs: #4398)

### PR DESCRIPTION
**Pull request by tk0miya**
refs: #4398 

Please correct my poor English.
(Alternative PR is welcome!)

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -92,12 +92,13 @@ These are the basic steps needed to start developing on Sphinx.
...
    For changes that should be included in the next minor release (namely bug
-   fixes), use the ``stable`` branch. ::
+   fixes), use the ``X.Y`` branch. ::
 
-       git checkout stable
+       git checkout X.Y
 
    For new features or other substantial changes that should wait until the
-   next major release, use the ``master`` branch.
+   next major release, use the ``master`` branch  (see `Branch Model`_ in
```
Comment (stephenfin) [inline CONTRIBUTING.rst]: (see Branch model **for details**)

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -187,7 +188,7 @@ These are the basic steps needed to start developing on Sphinx.
        git push origin feature-xyz
 
 #. Submit a pull request from your branch to the respective branch (``master``
-   or ``stable``) on ``sphinx-doc/sphinx`` using the GitHub interface.
+   or ``X.Y``) on ``sphinx-doc/sphinx`` using the GitHub interface.
```
Comment (stephenfin) [inline CONTRIBUTING.rst]: You can drop everything after the `)` here: if user's don't know how to submit a pull request, I doubt they'll be contributing.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -304,6 +305,22 @@ Debugging Tips
...
+
+Branch Model
+------------
+
+Sphinx project uses following branches for developing.
+
+``master``
+    Used for main development.  All improvement and refactoring, bugfixes are allowed.
+
+``X.Y`` (cf. 1.7)
```
Comment (stephenfin) [inline CONTRIBUTING.rst]: Let's try:

    ``X.Y``
      Where ``X.Y`` is the ``MAJOR.MINOR`` release. Used to maintain...

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -304,6 +305,22 @@ Debugging Tips
...
+    Used to maintain current stable release.  Only bugfixes and stable changes are allowed.
+    Basically, latest one major version is activated.
```
Comment (stephenfin) [inline CONTRIBUTING.rst]: Only the most recent stable release is currently retained. When a new version is released, the old release branch will be deleted and replaced by an equivalent tag.

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -304,6 +305,22 @@ Debugging Tips
...
+
+    .. note:: An old and unsupported versions are managed by tags.
```
Comment (stephenfin) [inline CONTRIBUTING.rst]: Drop this in favour of the above.

**Review by stephenfin — CHANGES_REQUESTED** [review]

**Comment by tk0miya** [comment]
Updated. Thank you for my English teacher :-)

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -61,8 +61,8 @@ of the core developers before it is merged into the main repository.
 #. If you feel uncomfortable or uncertain about an issue or your changes, feel
    free to email the *sphinx-dev* mailing list.
 #. Fork `the repository`_ on GitHub to start making your changes to the
-   **master** branch for next major version, or **stable** branch for next
-   minor version.
+   **master** branch for next major version, or **X.Y** branch for next
```
Comment (stephenfin) [inline CONTRIBUTING.rst]: One more nit: I see you use bold (\*\*) here and literals (\`\`) later on. Could you use literals here too?

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -304,6 +305,24 @@ Debugging Tips
...
+Branch Model
+------------
+
+Sphinx project uses following branches for developing.
+
+``master``
+    Used for main development.  All improvement and refactoring, bug fixes
+    are allowed.
+
+``X.Y`` (cf. 1.7)
```
Comment (stephenfin) [inline CONTRIBUTING.rst]: This one still looks off. What is `cf.`? Could you drop the brackets here, please

**Review by stephenfin — CHANGES_REQUESTED** [review]
Two more minor nits and we're good to go

File: CONTRIBUTING.rst (hunk 7 above)
Comment (tk0miya) [reply CONTRIBUTING.rst]: done

**Review by stephenfin — APPROVED** [review]
LGTM

**Comment by tk0miya** [comment]
Thanks!

**Pull request merged**